libcxx/utils/generate_feature_test_macro_components.py

```diff
@@ -190,15 +190,15 @@ def add_version_header(tc):
             "name": "__cpp_lib_atomic_wait",
             "values": {"c++20": 201907},
             "headers": ["atomic"],
-            "test_suite_guard": "!defined(_LIBCPP_AVAILABILITY_HAS_NO_SYNC)",
-            "libcxx_guard": "!defined(_LIBCPP_AVAILABILITY_HAS_NO_SYNC)",
+            "test_suite_guard": "!defined(_LIBCPP_AVAILABILITY_HAS_SYNC) || _LIBCPP_AVAILABILITY_HAS_SYNC",
```

Comment:
Like you suggested, I think `!defined(_LIBCPP_VERSION) || _LIBCPP_AVAILABILITY_HAS_SYNC` is better since it's more obvious why the check is written that way.